packages/aserto/src/aserto/client/directory/__init__.py

```diff
@@ -0,0 +1,450 @@
+            DeleteRelationRequest(param=relation_identifier), metadata=self._metadata
+        )
+
+    def check_relation(
+        self,
+        subject_type: str,
+        subject_key: str,
+        object_type: str,
+        object_key: str,
+        relation_type: str,
+    ) -> bool:
+        """Checks if a subject has a given relation to an object given
+        the object's type and key, the subject's type and key,
+        and relation type name.
+        Returns the result of the relation check, True or False.
+        object --relation_type--> subject
+
+        Parameters
+        ----
+        subject_type : str
+            the subject of the relation, a directory object type
+        subject_key : str
+            the subject of the relation, a directory object key
+        object_type : str
+            the object of the relation, a directory object type
+        object_key : str
+            the object of the relation, a directory object key
+        relation_type : str
+            a directory relation type
+
+        Returns
+        ----
+        True or False
+        """
+
+        response = self.reader.CheckRelation(
+            CheckRelationRequest(
+                object=ObjectIdentifier(type=object_type, key=object_key),
+                subject=ObjectIdentifier(type=subject_type, key=subject_key),
+                relation=RelationTypeIdentifier(name=relation_type, object_type=object_type),
+            ),
+            metadata=self._metadata,
+        )
+        return response.check
+
+    def check_permission(
+        self,
+        subject_type: str,
+        subject_key: str,
+        object_type: str,
+        object_key: str,
+        permission: str,
+    ) -> bool:
+        """Checks if a subject has a given permission on an object given
+        the object's type and key, the subject's type and key,
+        and permission name.
+        Returns the result of the permission check, True or False.
+
+        Parameters
+        ----
+        subject_type : str
+            the subject to check the permission for, a directory object type
+        subject_key : str
+            the subject to check the permission for, a directory object key
+        object_type : str
+            the object to check the permission on, a directory object type
+        object_key : str
+            the object to check the permission on, a directory object key
+        permission : str
+            a directory permission
+
+        Returns
+        ----
+        True or False
+        """
+
+        response = self.reader.CheckPermission(
+            CheckPermissionRequest(
+                object=ObjectIdentifier(type=object_type, key=object_key),
+                subject=ObjectIdentifier(type=subject_type, key=subject_key),
+                permission=PermissionIdentifier(name=permission),
+            ),
+            metadata=self._metadata,
+        )
+        return response.check
+
+    def close_channel(self) -> None:
```

Comment:
I'd rename this to just `close()`. If we had other cleanup to do besides closing the channel, we'd put it here too rather than have users call two separate `close_***()` functions.